pages/_app.js

```diff
@@ -14,13 +14,23 @@ export default function App({ Component, pageProps }) {
   function handleDeleteRecipe(id) {
     setRecipes(recipes.filter((recipe) => recipe.id !== id));
   }
+
+  function handleUpdateRecipe(updatedRecipe, id) {
+    const recipeIndex = recipes.findIndex((recipe) => recipe.id === id);
+    if (recipeIndex !== -1) {
+      recipes[recipeIndex] = updatedRecipe;
+      setRecipes(recipes);
+    }
+  }
```

Comment:
Here you update the recipes array directly and then use setRecipes to to change recipes to the recipes array you just changed it seems. I'm not sure if this is a way you want to handle the update process. Maybe you want to use a shallow copy instead:
```suggestion
  function handleUpdateRecipe(updatedRecipe) {
    setRecipes(
      recipes.map((recipe) => 
        recipe.id === updatedRecipe.id ? updatedRecipe : recipe
      )
    );
  }
```